From e3d323e664884003673a00559c14d8e799699b52 Mon Sep 17 00:00:00 2001
From: Petr Hluzin <email.is.fucking.required@invalid.invalid>
Date: Sat, 06 Oct 2012 12:45:34 +0000
Subject: AT90CAN128 has ELPM instruction.

According to AT90CAN32/64/128 datasheet rev.H all three devices in the
family have the ELPM instruction.

Reported by and patch provided by Uwe Bonnes, thanks.
http://savannah.nongnu.org/patch/?7850
---
(limited to 'src/at90canbase.cpp')

diff --git a/src/at90canbase.cpp b/src/at90canbase.cpp
index cbfef1a..da5b694 100644
--- src/at90canbase.cpp
+++ src/at90canbase.cpp
@@ -92,6 +92,7 @@ AvrDevice_at90canbase::AvrDevice_at90canbase(unsigned ram_bytes,
           &portf.GetPin(6),
           &portf.GetPin(7))
 {
+    flagELPMInstructions = true;
     flagJMPInstructions = (flash_bytes > 8U * 1024U) ? true : false;
     irqSystem = new HWIrqSystem(this, (flash_bytes > 8U * 1024U) ? 4 : 2, 37);
 
--
cgit v0.9.0.2
